dvc/repo/fetch.py

```diff
@@ -80,6 +80,8 @@ def outs_filter(out: "Output") -> bool:
 
             indexes[rev or "workspace"] = idx
         except Exception as exc:
+            if rev == "workspace" and rev not in revs:
+                continue
             if onerror:
                 onerror(rev, None, exc)
             collection_exc = exc
```

Comment:
Looking at the code, it seems we never raise error, unless there are no indexes.
So, it seems `dvc exp push` will only fail if both of the workspace and experiment ref failed, or if we have no experiment to push, but only the workspace.

Reply:
If there is an error only in the workspace, it will not fail, but it will loudly log an error message that looks like something went wrong. For example:

```
$ dvc exp push origin
ERROR: failed to collect 'workspace' - failed to parse 'stages.params_test.cmd' in 'dvc.yaml': Could not find 'params'
Pushed experiment rival-knap to Git remote 'origin'.
1 file uploaded
```

See also the test below.

Reply:
Are you going to also add that to `fetch` and `push`, please? I don't see it atm.

Reply:
Sorry, I meant to pass this `workspace=False|True` from `exp.push` down to `push` and then `fetch` and then to `_collect_indexes`. Otherwise, we'll skip workspace for all of them. 

The brancher is a legacy code and is too far away from many commands that I don't have an idea which command expect what when `all_branches`/`all_tags` etc are specified. Some may require workspace.

Reply:
This shouldn't impact brancher default behavior though since `workspace=True` by default. It should only impact `repo.fetch` behavior.

Reply:
Ahh, right. I did not look too closely there. :(

Reply:
What about `dvc exp push`?

Reply:
I looked where we are calling `fetch`, and I am fine with this PR, but I have to note we are changing behaviour here (i.e on `--all-commits` we are skipping workspace).

Reply:
It also uses this method.

Reply:
Right, I see that's also documented in https://dvc.org/doc/command-reference/push#-A, so I'll limit to `dvc exp push/pull` for now.

Reply:
Okay, limited it to exp push/pull